src/Parser.js

```diff
@@ -100,8 +102,28 @@ export function parse (value, root, parent, rule, rules, rulesets, pseudo, point
 				break
 			// :
 			case 58:
-				length = 1 + strlen(characters), property = previous
+				length = 1 + strlen(characters), property = previous, colon = caret()
 			default:
+				// (
+				if (character === 40) {
+					if (colon > 0) {
+						switch (slice(colon, caret() - 1)) {
+							case 'matches':
+							case 'is':
+							case 'not':
+							case 'any':
+							case 'where':
+							case 'has':
```

Comment:
any particular reason why do we have to maintain the allowlist here? couldn't we just assume that this should always be treated as a pseudoselector?

Reply:
I think that's a great question. I think my primary reason for doing it this way was to minimize the surface area of my change. Perhaps it's not necessary -- would require some more thought (and likely some more test cases) about how other css rules would possibly be impacted.

Reply:
A quick test removing the allowlist causes issues with `calc` and `url`:

```
height:calc( 100vh - 1px );
background:url("data:image/png;base64,iVBORw0");
```

Reply:
Those are mostly used in the declarations - however, they can also appear in the selectors~. `calc` can be used within `@media` and `url` can be used together with `@document`. It seems to me though that for them to both appear in the selectors~ and be preceded by a colon is not likely at all - or am I missing something? I wonder if this could be used somehow here. 🤔

Reply:
I agree, the likelihood is low and we could probably simplify.

Right now my logic is basically:

```
if we saw a "(" preceded by ":matches", ":is", etc.
    continue parsing selector character by character
else
   skip parsing entire "(...)" block
```

What the logic could potentially be simplified to is:

```
if we saw a "(" and we're currently parsing a selector
    continue parsing selector character by character
else // we're parsing a declaration
   skip parsing entire "(...)" block
```

I think this is where I'm limited by my lack of familiarity with the code. Are we able to determine whether or not we are currently parsing a selector or a declaration?

Reply:
any thoughts on this @thysultan ?